py: Add LIBS_USERMOD to LIBS.

sfe-SparkFro opened the pull request:
### Summary

Progress towards #18351

This enables C++ modules to correctly postion `-l` linker flags at the end of the flags instead of at the start. Updated the example C++ `micropython.mk` accordingly.

### Testing

Attempted to build firmware for a board that uses Make (eg. `stm32` or `mimxrt`). Before this change, the linker could not find the symbols from `libstdc++.a`. After this change, those symbols were found.

sfe-SparkFro committed:
py: Add LIBS_USERMOD to LIBS.

This enables C++ modules to correctly postion -l
linker flags at the end of the flags instead of at
the start. Updated the example C++ micropython.mk
accordingly.

Signed-off-by: Dryw Wade <dryw.wade@sparkfun.com>

dpgeorge commented:
> Would it be acceptable to simply add `LDFLAGS += $(LIBS)` after the last modification of `LDFLAGS`? That makes it build successfully on my end.

The unix and Windows ports use `LIB`... and that's inconsistent with all other ports which use `LIBS`.

So I think the right solution here is to change the two instances of `LIB` in `py/mkrules.mk` and `ports/windows/Makefile` to `LIBS`.

sfe-SparkFro committed:
py/mkrules: Change LIB to LIBS.

Same in ports/windows/Makefile to be consistent
with other ports. Resolves build error after 7176acb.

Signed-off-by: Dryw Wade <dryw.wade@sparkfun.com>

sfe-SparkFro commented:
dbaa315 passes :tada:

dpgeorge commented:
Rebased and merged in 01df5a1f0512fdabd8d40e465b07eb12f0573744 and 234bac45d33f410fb3b56a87cf56266e96a71c66
